Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/configuration/CCLConfigurationModule.kt

```diff
@@ -0,0 +1,55 @@
+package de.rki.coronawarnapp.ccl.configuration
+
+import android.content.Context
+import dagger.Module
+import dagger.Provides
+import de.rki.coronawarnapp.ccl.configuration.server.CCLConfigurationApiV1
+import de.rki.coronawarnapp.environment.download.DownloadCDNHttpClient
+import de.rki.coronawarnapp.environment.download.DownloadCDNServerUrl
+import de.rki.coronawarnapp.util.di.AppContext
+import okhttp3.Cache
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import java.io.File
+import javax.inject.Qualifier
+import javax.inject.Singleton
+
+@Module
+object CCLConfigurationModule {
+
+    @Singleton
+    @CCLConfiguration
+    @Provides
+    // Holds the ccl and ccl http cache. Gets cleared in CCLConfigurationRepository
+    fun provideCCLDir(@AppContext context: Context): File = File(context.filesDir, CCL_DIR)
+
+    @Singleton
+    @Provides
+    fun provideApi(
+        @DownloadCDNHttpClient client: OkHttpClient,
+        @DownloadCDNServerUrl url: String,
+        @CCLConfiguration cclDir: File
+    ): CCLConfigurationApiV1 {
+        val cacheDir = File(cclDir, CCL_CACHE_DIR)
+        val cache = Cache(cacheDir, CACHE_SIZE)
+
+        val cclClient = client.newBuilder()
+            .cache(cache)
+            .build()
+
+        return Retrofit.Builder()
+            .client(cclClient)
+            .baseUrl(url)
+            .build()
+            .create(CCLConfigurationApiV1::class.java)
+    }
+}
+
+private const val CCL_DIR = "ccl"
+private const val CCL_CACHE_DIR = "ccl_config_http_cache"
+private const val CACHE_SIZE = 5 * 1024 * 1024L // 5MB
```

Comment:
Can you align with Max whether 5 MB is enough? afaik he once used a file of 7 MB size

Reply:
Do we actually need a http cache? We store the config in a file anyway?

Reply:
I think we need to define the cache so that the ETag handling works with okhttp....

Reply:
yes cache was wished. I raised the cache size to be on the safe side.